controllers/component_build_controller_unit_test.go

```diff
@@ -298,27 +299,51 @@ func TestGenerateInitialPipelineRunForComponentDockerfileContext(t *testing.T) {
 		return &dockerfileImage, nil
 	}
 
-	pipelineRun, err := generatePipelineRunForComponent(component, pipelineRef, additionalParams, &buildGitInfo{})
-
-	if err != nil {
-		t.Error("generateInitialPipelineRunForComponentDockerfileContext(): Failed to generate pipeline run")
+	tests := []struct {
+		name              string
+		pipelineLabelUsed bool
```

Comment:
Please avoid it. Use condition based on component or even duplicate the tests which will make removal trivial operation.

Reply:
changed

I am copy pasting tests which use build pipeline selector
so we can just easily remove them later